### ISSUE TITLE
Feature: to_string(const json& j);

### ISSUE THREAD

**Issue by DELTA37** [nlohmann/json#916]
It is very helpful to add method to_string(const json& j);

**Comment by theodelrieu** [comment]
Agreed, would you like to open a PR? This one seems quite easy to implement :)

**Comment by nlohmann** [comment]
What would be the expected result? `dump()`?

**Comment by theodelrieu** [comment]
That's what I would expect, do you agree @DELTA37?

**Comment by cronnosli** [comment]
I guess that .dump() is already enough, unless the intent is to replace .dump()!

**Comment by theodelrieu** [comment]
The use-case I see would be something like the following:
 
```cpp
// in some generic code
template <typename T>
void doStuff(T const& value) {
    using std::to_string;
    
    auto const str = to_string(value);
    // process the string
}
```

Other than that, I agree that `dump()` is pretty much enough.

**Comment by DELTA37** [comment]
Yes, i guess to add "to_string()" method which will do the same as dump()
It is a standard method to extend std::to_string for user-defined class
And it is very helpful to do it with json

That is a discussion https://stackoverflow.com/questions/33399594/making-a-user-defined-class-stdto-stringable

// Sorry for my English

**Comment by DELTA37** [comment]
https://github.com/nlohmann/json/pull/918
Attach pill request

**Comment by Macr0Nerd** [comment]
@nlohmann is this still available? It doesn't seem terribly important but it was marked as good first issue so :)

**Comment by nlohmann** [comment]
@Macr0Nerd Yes, it is! There were open issues in PR #918, so this issue was never fixed.

**Comment by Macr0Nerd** [comment]
Alright! So @nlohmann all I need to is read a function that takes the JSON key, and pulls a deserializes the data into a key, correct? I’m still relatively green to C++ but I’ll give this my best!

**Comment by nlohmann** [comment]
No, a `json` value is given and `to_string` should return its serialization.

**Comment by Macr0Nerd** [comment]
So essentially just do what .dump() does and return the entire given JSON object passed as a string? I apologize if I am asking to many questions, I just want to confirm that i correctly do it.

**Comment by Macr0Nerd** [comment]
I think i see what you want (and based on the previous responses). It seems that you want to add an argument dependent lookup like [this comment](https://stackoverflow.com/a/33399769/5859623) from the SO post from above. As it wasn't requested, I wouldn't overload the `<<` operator but is this what you want @nlohmann?

I also apologize if I ask too many questions, I'm just of the mentality that it should be done right the first time (for the most part)

**Comment by Macr0Nerd** [comment]
@nlohmann I believe I understand the purpose of this feature. It seems as if though the logic (and code) of dump would do the job here. I want to know if this is alright and what you want/expect

**Comment by Macr0Nerd** [comment]
Well for lack of understanding and the fear of messing up a very useful product, I will be removing myself from this problem @nlohmann

**Comment by nlohmann** [comment]
Basically look at https://github.com/nlohmann/json/pull/918/files and follow the comments.

**Comment by Macr0Nerd** [comment]
@nlohmann I put the to_string in the the same place as #918, but I saw that there needed to be some unit testing. Which file would I put the tests in?

**Comment by nlohmann** [comment]
Closed by merging #1585.
